src/OpenLoco/Town.cpp

```diff
@@ -13,12 +13,29 @@ namespace OpenLoco
         return name == StringIds::null;
     }
 
-    // 0x0049742F
+    /**
+     * 0x0049742F
+     * Update town
+     *
+     * @param this @<esi>
+     */
     void Town::update()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
-        call(0x0049742F, regs);
+        recalculateSize();
+
+        static const std::array<uint8_t, 12> buildSpeedToGrowthPerTick = { 0, 1, 3, 5, 7, 9, 12, 16, 22, 0, 0, 0 };
+        auto buildSpeed = buildSpeedToGrowthPerTick[this->build_speed];
+        if (buildSpeed == 0 || (buildSpeed == 1 && (gPrng().randNext() & 7) == 0))
```

Comment:
this should be `buildSpeed == 0 || (buildSpeed == 1 && (gPrng().randNext() & 7))`